New pages: HTMLMediaElement API (played and seeking) (#37006)
mdn/content: files/en-us/web/api/htmlmediaelement/played/index.md

---
title: "HTMLMediaElement: played property"
short-title: played
slug: Web/API/HTMLMediaElement/played
page-type: web-api-instance-property
browser-compat: api.HTMLMediaElement.played
---

{{APIRef("HTML DOM")}}

> The **`played`** read-only property of the {{domxref("HTMLMediaElement")}} interface indicates the time ranges the resource, an {{htmlelement("audio")}} or {{htmlelement("audio")}} media file, has played. It returns a new {{domxref("TimeRanges")}} object that contains the ranges of the media source that the browser has played, if any, at the time the attribute is evaluated.

Author: ```suggestion
The **`played`** read-only property of the {{domxref("HTMLMediaElement")}} interface indicates the time ranges the resource, an {{htmlelement("audio")}} or {{htmlelement("video")}} media file, has played. It returns a new {{domxref("TimeRanges")}} object that contains the ranges of the media source that the browser has played, if any, at the time the attribute is evaluated.
```

## Value

> A {{domxref("TimeRanges")}} object, which includes the number of ranges as a `length`, with the `start` and `end` time for each range.

Author: ```suggestion
A {{domxref("TimeRanges")}} object; representing the time ranges that have been played.
```

---
title: "HTMLMediaElement: played property"
short-title: played
slug: Web/API/HTMLMediaElement/played
page-type: web-api-instance-property
browser-compat: api.HTMLMediaElement.played
---

{{APIRef("HTML DOM")}}

> The readonly **`played`** property of the {{domxref("HTMLMediaElement")}} interface indicates the time ranges the media, an {{htmlelement("audio")}} or {{htmlelement("audio")}} media file, has played. It returns a new {{domxref("TimeRanges")}} object that contains the ranges of the media source that the browser has played, if any, at the time the attribute is evaluated.

Reviewer: We use "read-only" and AFAIK the usual wording here is "The $NAME read-only property of the $INTERFACE interface".

Reviewer: > the media, an {{htmlelement("audio")}} or {{htmlelement("audio")}} media file

"the media" here is ungrammatical IMO. Could be "the source" or "the resource". Also, "audio or audio file"?

I would have a para break before "It returns".

> The readonly **`played`** property of the {{domxref("HTMLMediaElement")}} interface indicates the time ranges the media, an {{htmlelement("audio")}} or {{htmlelement("audio")}} media file, has played. It returns a new {{domxref("TimeRanges")}} object that contains the ranges of the media source that the browser has played, if any, at the time the attribute is evaluated.

Author: ```suggestion
The **`played`** read-only property of the {{domxref("HTMLMediaElement")}} interface indicates the time ranges the resource, an {{htmlelement("audio")}} or {{htmlelement("audio")}} media file, has played. It returns a new {{domxref("TimeRanges")}} object that contains the ranges of the media source that the browser has played, if any, at the time the attribute is evaluated.
```

## Value

> A {{domxref("TimeRanges")}} object, which includes the number of ranges as a `length`, with the `start` and `end` time for each range.

Reviewer: I don't think code-formatting `start` and `end` is appropriate since they're not actually properties. TBH I think you could just say "A  {{domxref("TimeRanges")}} object representing the time ranges that have been played." and leave the specifics to the linked page, which someone is is going to have to read anyway if they are to use this property.

## Examples

```js
const media = document.querySelector("audio");
const playedTimeRanges = media.played;
let timePlayed = 0;
// calculate the total time the media has played
for (let i = 0; i < playedTimeRanges.length; i++) {
  timePlayed += playedTimeRanges.end(i) - playedTimeRanges.start(i);
}
console.log(`The media played for a total of ${timePlayed} seconds.`);

> ```

Reviewer: would be great to see this as a live sample.

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

- {{domxref("HTMLMediaElement.seeked_event", "seeked")}} event
- {{domxref("HTMLMediaElement.progress_event", "progress")}} event
- {{domxref("HTMLMediaElement.seekable")}}
- {{domxref("HTMLMediaElement.buffered")}}

> - {{domxref("HTMLMediaElement")}}

Reviewer: I don't think we should link to the parent here.

> - {{domxref("HTMLMediaElement.buffered")}}
> - {{domxref("HTMLMediaElement")}}

Author: ```suggestion
- {{domxref("HTMLMediaElement.buffered")}}
```
